src/core/server/csp/config.test.ts

```diff
@@ -9,11 +9,465 @@
 import { config } from './config';
 
 describe('config.validate()', () => {
-  test(`does not allow "disableEmbedding" to be set to true`, () => {
+  it(`does not allow "disableEmbedding" to be set to true`, () => {
     // This is intentionally not editable in the raw CSP config.
     // Users should set `server.securityResponseHeaders.disableEmbedding` to control this config property.
     expect(() => config.schema.validate({ disableEmbedding: true })).toThrowError(
       '[disableEmbedding]: expected value to equal [false]'
     );
   });
+
+  describe(`"script_src"`, () => {
+    it(`throws if containing 'unsafe-inline' when 'strict' is true`, () => {
+      expect(() =>
+        config.schema.validate({
+          strict: true,
+          script_src: [`'self'`, `unsafe-inline`],
+        })
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp.strict\` is true"`
+      );
+
+      expect(() =>
+        config.schema.validate({
+          strict: true,
+          script_src: [`'self'`, `'unsafe-inline'`],
+        })
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp.strict\` is true"`
+      );
+    });
+
+    it(`throws if containing 'unsafe-inline' when 'warnLegacyBrowsers' is true`, () => {
+      expect(() =>
+        config.schema.validate({
+          warnLegacyBrowsers: true,
+          script_src: [`'self'`, `unsafe-inline`],
+        })
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp.strict\` is true"`
+      );
+
+      expect(() =>
+        config.schema.validate({
+          warnLegacyBrowsers: true,
+          script_src: [`'self'`, `'unsafe-inline'`],
+        })
+      ).toThrowErrorMatchingInlineSnapshot(
+        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp.strict\` is true"`
+      );
+    });
```

Comment:
Whoops, looks like this was copied from the other test case but is missing `strict: false`

```suggestion
    it(`throws if containing 'unsafe-inline' when 'warnLegacyBrowsers' is true`, () => {
      expect(() =>
        config.schema.validate({
          strict: false,
          warnLegacyBrowsers: true,
          script_src: [`'self'`, `unsafe-inline`],
        })
      ).toThrowErrorMatchingInlineSnapshot(
        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp. warnLegacyBrowsers\` is true"`
      );

      expect(() =>
        config.schema.validate({
          strict: false,
          warnLegacyBrowsers: true,
          script_src: [`'self'`, `'unsafe-inline'`],
        })
      ).toThrowErrorMatchingInlineSnapshot(
        `"cannot use \`unsafe-inline\` for \`script_src\` when \`csp. warnLegacyBrowsers\` is true"`
      );
    });
```